src/agentbeats/tool_provider.py

```
@@ -0,0 +1,27 @@
+
+        Args:
+            message: The message to send to the agent
+            url: The agent's URL endpoint
+            new_conversation: If True, start fresh conversation; if False, continue existing conversation
+
+        Returns:
+            str: The agent's response message
+        """
+        outputs = await send_message(message=message, base_url=url, context_id=None if new_conversation else self._context_ids.get(url, None))
+        if outputs.get("status", "completed") != "completed":
+            raise RuntimeError(f"{url} responded with: {outputs}")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)